src/Main.tsx

```diff
@@ -7,56 +7,84 @@ import { Infos } from './basicElements/infos';
+import { Splash } from './basicElements/splash';
+import { CurrentUser } from "./basicElements/user-hooks/types";
+
 
 function Main() {
   const [info, setInfo] = useState(true);
   const [team, setTeam] = useState({ uuid: '', email: '', name: '' } as Team);
   const [exercise, setExercise] = useState({} as Exercise); //{uuid: 'dsaads', sequence_number: 0, category_ord: 0, category_uuid: '', description:'dasdads asdads', title:'dasasd', max_points: 3}
   const [category, setCategory] = useState({} as Category);
-  const tokenGetter= new URLSearchParams(useLocation().search);
+  const tokenGetter = new URLSearchParams(useLocation().search);
   const token = tokenGetter.get("one-time-auth")
   const authHeader = useAuthHeader();
   const user = useCurrentUser();
   useEffect(() => {
     if (user) {
       getTeam(authHeader).then(team => {
-        if(team!==null)
+        if (team !== null)
           setTeam(team)
       });
       getCategory(authHeader, user.categoryUuid).then(cat => {
-        if(cat!==null)
+        if (cat !== null)
           setCategory(cat)
       });
       getCurrentExercise(authHeader).then(exc => {
-        if(exc!==null)
+        if (exc !== null)
           setExercise(exc);
       });
     }
   }, [user]);
-  if(token){
+  if (token) {
     return (
       <Layout>
-        <OneTimeTokenLogin token = {token}/>
+        <OneTimeTokenLogin token={token} />
       </Layout>
     )
   }
+  const inProgress = exercise?.category_ord > 0;
+  const type = getReactComponent(user, category, inProgress, team, info);
   return (
     <Layout>
-      <GoogleAnalytics identifier={'G-SZ3DY6CGPS'}/>
-      <LoadUserOnClientSide/>
-      {!user && <Login />}
-      {user && info && <Infos exercise={exercise} setExercise={setExercise} teamFinished={category.starts_at?.isBefore(moment.now())} teamInProgress={exercise?.category_ord > 0} setInfo={setInfo} teamName={team.name} categoryName={category.name} categoryEnd={category.ends_at} categoryStart={category.starts_at}/>}
-      {user && !info && <Excercise auth={authHeader} exercise={exercise} endsAt={category?.ends_at} teamName={team.name} setInfo={setInfo}/>}
+      <GoogleAnalytics identifier={'G-SZ3DY6CGPS'} />
+      <LoadUserOnClientSide />
+      {type===MainComponentType.Login && <Login />}
+      {type===MainComponentType.Splash && <Splash team={team} setTeam={setTeam} setExcercise={setExercise} inProgress={inProgress}/>}
+      {type===MainComponentType.Infos && <Infos exercise={exercise} setExercise={setExercise} teamFinished={category.global_ends_at?.isBefore(moment.now())} teamInProgress={inProgress} setInfo={setInfo} teamName={team.name} categoryName={category.name} categoryEnd={category.global_ends_at || null} categoryStart={category.global_starts_at || null} />}
+      {type===MainComponentType.Excercise && <Excercise auth={authHeader} exercise={exercise} endsAt={team?.ends_at || null} teamName={team.name} setInfo={setInfo} />}
```

Comment:
![image](https://user-images.githubusercontent.com/7879888/138102120-4b839f14-5df0-4389-ad68-26599a75d8bc.png)

Oh, this uses null. Maybe it is better to use undefined instead of null...